### PR TITLE
feat: added avatar decoration url to the dpp::user class

### PR DESCRIPTION
**Pull request by Commandserver**
https://github.com/discord/discord-api-docs/commit/3074b5fa04658b9204f60fd6cd0a0eb151bcb67b

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

File: src/dpp/user.cpp [hunk 1]
```diff
@@ -126,6 +126,16 @@ std::string user::get_default_avatar_url() const {
 	}
 }
 
+std::string user::get_avatar_decoration_url(uint16_t size) const {
+	if (this->id) {
+		return utility::cdn_endpoint_url_hash({ i_png },
+											  "avatar-decorations/" + std::to_string(this->id), this->avatar_decoration.to_string(),
```
Comment (braindigitalis) [inline src/dpp/user.cpp:132]: i heard you like indents so i put indents in your indents in your indents in...

Comment (Commandserver) [reply src/dpp/user.cpp:132]: 🤣 wanted to stick with the existing ones

**Comment by Commandserver** [comment]
not sure if we want this feature at all. It's currently in beta. I wonder why this feature is even documented, but hey it's discord...
Also:
> The available selection of avatar decorations will change periodically, and can be used by Nitro Subscribers for a limited time only.

I think no one is seriously using this in their bots.

**Comment by braindigitalis** [comment]
> not sure if we want this feature at all. It's currently in beta. I wonder why this feature is even documented, but hey it's discord... Also:
> 
> > The available selection of avatar decorations will change periodically, and can be used by Nitro Subscribers for a limited time only.
> 
> [support.discord.com/hc/en-us/articles/13410113109911-Avatar-Decorations](https://support.discord.com/hc/en-us/articles/13410113109911-Avatar-Decorations) I think no one is seriously using this in their bots. At least not yet

If its documented, its staying around, lets include it. Is this ready to merge yet?

**Comment by Commandserver** [comment]
> If its documented, its staying around, lets include it. Is this ready to merge yet?

It's ready but untested

File: include/dpp/user.h [hunk 2]
```diff
@@ -93,6 +93,8 @@ class DPP_EXPORT user : public managed, public json_interface<user>  {
 	std::string global_name;
 	/** Avatar hash */
 	utility::iconhash avatar;
+	/** Avatar decoration hash */
+	utility::iconhash avatar_decoration;
 	/** Flags built from a bitmask of values in dpp::user_flags */
```
Comment (braindigitalis) [inline include/dpp/user.h:98]: should this be here or under user_identified?
Are the avatar_decoration sent via the websocket?

Comment (Commandserver) [reply include/dpp/user.h:98]: that's a good question. Idk. can't test that either as it's not out yet 😥

**Review by braindigitalis — CHANGES_REQUESTED** [review]
will change to approved once we know how it works

**Comment by braindigitalis** [comment]
it's been two months, is this happening or did discord shitcan the fearure?

**Comment by Jaskowicz1** [comment]
https://discord.com/developers/docs/resources/user#user-object It exists, and mishura used to use it too. Might be worth testing?

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**